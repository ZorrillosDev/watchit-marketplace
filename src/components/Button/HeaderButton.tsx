import React, { FC } from 'react'
import { ButtonProps } from '@material-ui/core'
import { TabletAndDown, TabletAndUp } from '@styles/breakpoints'
import styled from 'styled-components'
import RoundButton from '@components/Button/RoundButton'

interface HeaderButtonProps {
  text: string
  mobileText: string
}

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
const HeaderButton: FC<HeaderButtonProps & ButtonProps> = ({ text, mobileText, ...props }): JSX.Element => {
  return (
    <HeaderButtonStyled {...props}>
      <TabletAndUp> {text} </TabletAndUp>
      <TabletAndDown> {mobileText} </TabletAndDown>
    </HeaderButtonStyled>
  )
}

export default HeaderButton

export const HeaderButtonStyled = styled(RoundButton)`
  border-style: solid;
  margin-left: 0.5rem;
  border-radius: 0.5rem;
  padding: 6px 1.5rem;
  height: 2rem;
  box-shadow: 2px 2px 8px rgba(0,0,0,0.15);

  span {
    margin-top: 4px;
  }

  @media (max-width: 640px) {
    padding: 6px 1rem;
  }
  
  @media (max-width: 380px) {
    padding: 6px 0.5rem;
  }
`
