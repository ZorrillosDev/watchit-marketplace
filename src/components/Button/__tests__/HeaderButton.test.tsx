import HeaderButton, { HeaderButtonStyled } from '@components/Button/HeaderButton'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'

/* eslint-disable no-undef */
/* eslint-disable @typescript-eslint/consistent-type-assertions */
describe('<HeaderButton />', () => {
  it('should render', () => {
    const component = shallow(<HeaderButton text='test text' mobileText='mobile test text' />)
    expect(component).toMatchSnapshot()
  })

  it('should render header button props text', () => {
    const titleText = 'Test Module'
    const component = shallow(<HeaderButton text={titleText} mobileText='mobile test text' />)

    expect(component.text()).toContain(titleText)
  })

  it('should have header button valid styles', () => {
    const component = shallow(<HeaderButtonStyled />)

    expect(component).toHaveStyleRule('box-shadow', '2px 2px 8px rgba(0,0,0,0.15)')
    expect(component).toHaveStyleRule('padding', '6px 1.5rem')
    expect(component).toHaveStyleRule('padding', '6px 0.5rem', { media: '(max-width: 380px)' })
  })
})
