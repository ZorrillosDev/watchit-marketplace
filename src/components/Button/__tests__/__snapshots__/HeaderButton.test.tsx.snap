// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderButton /> should render 1`] = `ShallowWrapper {}`;
